rpcs3/Emu/RSX/rsx_methods.cpp

```diff
@@ -876,16 +876,22 @@ namespace rsx
 					{
 						// Move last 32 bits
 						reinterpret_cast<u32*>(dst)[0] = reinterpret_cast<const u32*>(src)[count - 1];
-					}
-					else if (dst_dma & CELL_GCM_LOCATION_MAIN)
-					{
-						// May overlap
-						std::memmove(dst, src, data_length);
+						rsx->invalidate_fragment_program(dst_dma, dst_offset, 4);
```

Comment:
Why is this method call being repeated inside the switch when it can be called once outside?